### PR TITLE
Fix GeneralFilter::matchPacketWithFilter to  match packet of any link layer. fixes seladb/PcapPlusPlus#498

### PR DESCRIPTION
**Pull request by skorup101**
Also added few tests.

### CODE REVIEW

File: Pcap++/header/PcapFilter.h [hunk 1]
```diff
@@ -80,6 +81,7 @@ namespace pcpp
 	protected:
 		bpf_program* m_program;
 		std::string m_lastProgramString;
+		pcpp::LinkLayerType m_lastLinkLayerType;
```
Comment (seladb) [inline Pcap++/header/PcapFilter.h:84]: Please change the variable name to `m_LastLinkLayerType` (with capital L).

The other variable names should be changed as well, but me or you can do it later.

File: Pcap++/header/PcapFilter.h [hunk 2]
```diff
@@ -80,6 +81,7 @@ namespace pcpp
 	protected:
 		bpf_program* m_program;
 		std::string m_lastProgramString;
```
Comment (seladb) [inline Pcap++/header/PcapFilter.h:83]: Can you also change these variable names to begin with `m_<capital letter>`?

File: Pcap++/header/PcapFilter.h [hunk 3]
```diff
@@ -118,7 +120,7 @@ namespace pcpp
 		const std::string m_filterStr;
 
 	public:
-		BPFStringFilter(const std::string& filterStr) : m_filterStr(filterStr) {}
+		BPFStringFilter(const std::string& filterStr) : m_filterStr(filterStr) {}
```
Comment (seladb) [inline Pcap++/header/PcapFilter.h:123]: was there any change here?

File: Tests/Pcap++Test/TestDefinition.h [hunk 4]
```diff
@@ -35,6 +35,7 @@ PTF_TEST_CASE(TestPcapFilters_matchPacketWithFilter_static);
 PTF_TEST_CASE(TestPcapFiltersLive);
 PTF_TEST_CASE(TestPcapFilters_General_BPFStr);
 PTF_TEST_CASE(TestPcapFiltersOffline);
+PTF_TEST_CASE(TestPcapFilters_matchPacketWithFilter_DifferentInterfaces);
```
Comment (seladb) [inline Tests/Pcap++Test/TestDefinition.h:38]: Please change the test name to fit 30 chars or less

File: Tests/Pcap++Test/Tests/FilterTests.cpp [hunk 5]
```diff
@@ -736,4 +735,88 @@ PTF_TEST_CASE(TestPcapFiltersOffline)
...
+	PTF_ASSERT_TRUE(fileReaderDev1.open());
+	pcpp::RawPacketVector rawPacketVec;
+	fileReaderDev1.getNextPackets(rawPacketVec);
+	fileReaderDev1.close();
+
+	int validCounter = 0;
+	for (pcpp::RawPacketVector::VectorIterator iter = rawPacketVec.begin(); iter != rawPacketVec.end(); iter++)
+	{
+		pcpp::Packet packet(*iter);
+		if(pcpp::IPv4Layer* l = packet.getLayerOfType<pcpp::IPv4Layer>())
```
Comment (seladb) [inline Tests/Pcap++Test/Tests/FilterTests.cpp:755]: please change the variable name from `l` to something with more meaning like `ip4Layer`

File: Pcap++/header/PcapFilter.h (hunk 1 above)
Comment (skorup101) [reply Pcap++/header/PcapFilter.h]: Done.

File: Pcap++/header/PcapFilter.h (hunk 2 above)
Comment (skorup101) [reply Pcap++/header/PcapFilter.h]: Done.

File: Pcap++/header/PcapFilter.h (hunk 3 above)
Comment (skorup101) [reply Pcap++/header/PcapFilter.h]: Nope. It's probably different line ending.

File: Tests/Pcap++Test/TestDefinition.h (hunk 4 above)
Comment (skorup101) [reply Tests/Pcap++Test/TestDefinition.h]: Done.

File: Tests/Pcap++Test/Tests/FilterTests.cpp (hunk 5 above)
Comment (skorup101) [reply Tests/Pcap++Test/Tests/FilterTests.cpp]: Done.

**Pull request merged**